src/class2_recursion_binarysearch/SearchInSortMatrix1.java

```diff
@@ -1,48 +1,70 @@
 package class2_recursion_binarysearch;
 
-public class SearchInSortMatrix1 {
-    /*
-    Apple binary search in 2D space
-    Variant(变种)1.0 application： 2D matrix, sorted on each row, first element
-    of next row is larger (or equal) to the last element of previous row, now giving a target number,
-    returning hte position that the target locates within the matrix
-    二维数组找target
-    1    2    3    4
-    5    6    7    8
-    9    10   11   12
+/**
+ * /**
+ *  * Description
+ *  * Given a 2D matrix that contains integers only, which each row is sorted in an ascending order. The first element of next row is larger than (or equal to) the last element of previous row.
+ *  *
+ *  * Given a target number, returning the position that the target locates within the matrix. If the target number does not exist in the matrix, return {-1, -1}.
+ *  *
+ *  * Assumptions:
+ *  *
+ *  * The given matrix is not null, and has size of N * M, where N >= 0 and M >= 0.
+ *  * Examples:
+ *  * col = 3 , row = 4
+ *  * matrix = { {1, 2, 3}, {4, 5, 7}, {8, 9, 10} , {11 , 12 , 15} }
+ *  *
+ *  * target = 7, return {1, 2}
+ *  *
+ *  * target = 6, return {-1, -1} to represent the target number does not exist in the matrix.
+ *  *
+ *  * 关于array
+ *  * int[] array = new int[5];   表示array是长度为5的数组 这是初始化
+ *  * array[2] = 6; 表示array中第2个数是6
+ *  */
 
-    target == 6？？？？
-     */
+/*
+   Apple binary search in 2D space
+   Variant(变种)1.0 application： 2D matrix, sorted on each row, first element
+   of next row is larger (or equal) to the last element of previous row, now giving a target number,
+   returning hte position that the target locates within the matrix
+   二维数组找target
+   1    2    3    4
+   5    6    7    8
+   9    10   11   12
+
+   target == 6？？？？
+    */
     /*
     method 1
     Step 1: 先竖着找列，在m列中找到最接近target的num。确定行数 O(log(n))
     Step 2: 确定了target所在列数,再在n行里找target所在的列 O(log(m))
     Time = O(log(m) + log(n)) = O(log(mn))
      */
-
-    /*
+     /*
     method 2
```

Comment:
Your method 2 is wrong, delete for now until Leetcode same question is resolved

Reply:
done